# HG changeset patch
# User Rik <octave@nomad.inbox5.com>
# Date 1318796732 25200
#      Sun Oct 16 13:25:32 2011 -0700
# Node ID a0875c6cab7e9dd27ff8cbfc3f24930b50afaefa
# Parent  ee28bf4c3c292322d0ee38e3436aacb0d0fa54ba
error.cc: Don't complain if error struct input is empty (Matlab compatability)

* error.cc (error): Don't complain if error struct input is empty
(Matlab compatability)

diff --git a/src/error.cc b/src/error.cc
--- a/src/error.cc
+++ b/src/error.cc
@@ -1057,16 +1057,20 @@ error: nargin != 1\n\
                     nargs(i-1) = args(i);
                 }
             }
           else
             return retval;
         }
       else if (nargin == 1 && args(0).is_map ())
         {
+          // empty struct is not an error.  return and resume calling function.
+          if (args(0).is_empty ())
+            return retval;
+
           octave_value_list tmp;
 
           octave_scalar_map m = args(0).scalar_map_value ();
 
           // empty struct is not an error.  return and resume calling function.
           if (m.nfields () == 0)
             return retval;
 
